src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -268,7 +268,10 @@ internal bool IsClosedLocked
             }
         }
 
-        internal Collection DefaultCollection => _defaultCollection == null ? GetDefaultCollection() : _defaultCollection;
+        internal Collection DefaultCollection => _defaultCollectionIsDeleted || 
```

Comment:
There is still a risk here that I don't like.  If you call this before you ever call `GetDefaultCollection()` then this is still going to return null because that is what `GetDefaultCollection()` returns.

My suggestion is to look at the result of `GetDefaultCollection()` and if it is null then throw the exception.  

```
if(_defaultCollection == null) {
    GetDefaultCollection();
}

if(_defaultCollection == null || _defaultCollection.IsClosed) {
    throw new InvalidOperationException($"Default Collection is deleted from the database {ToString()}.")
}

return _defaultCollection;
```